### PR TITLE
Change Coin to (de)serailize from and to string(CRO-125)

### PR DESCRIPTION
**Pull request by leejw51crypto**
this is for client rpc coin serialization
keep Coin as originally,
and use String just communicating with angular side

### CODE REVIEW

File: chain-core/src/init/coin.rs [hunk 1]
```diff
@@ -52,6 +52,9 @@ impl ::std::error::Error for CoinError {}
 type CoinResult = Result<Coin, CoinError>;
 
 impl Coin {
+    pub fn value(&self) -> u64 {
```
Comment (tomtau) [inline chain-core/src/init/coin.rs]: Perhaps as `impl Into<u64> for Coin`?

File: client-common/src/tendermint/client.rs [hunk 2]
```diff
@@ -17,4 +18,7 @@ pub trait Client: Send + Sync {
 
     /// Makes `broadcast_tx_sync` call to tendermint
     fn broadcast_transaction(&self, transaction: &[u8]) -> Result<()>;
+
+    /// Get staked-state from the staked stake address
+    fn get_account(&self, staked_state_address: &[u8]) -> Result<StakedState>;
```
Comment (tomtau) [inline client-common/src/tendermint/client.rs]: seems this change isn't related to the coin deserialisation?

File: client-common/src/tendermint/rpc_client.rs [hunk 3]
```diff
@@ -65,4 +67,35 @@ impl Client for RpcClient {
         self.call::<serde_json::Value>("broadcast_tx_sync", &params)
             .map(|_| ())
     }
+
+    fn get_account(&self, staked_state_address: &[u8]) -> Result<StakedState> {
```
Comment (tomtau) [inline client-common/src/tendermint/rpc_client.rs]: not related to coin serialisation

File: client-index/src/index/default_index.rs [hunk 4]
```diff
@@ -469,6 +470,16 @@ mod tests {
         fn broadcast_transaction(&self, _: &[u8]) -> Result<()> {
             Ok(())
         }
+
+        fn get_account(&self, staked_state_address: &[u8]) -> Result<StakedState> {
```
Comment (tomtau) [inline client-index/src/index/default_index.rs]: not related to coin serialisation

File: client-network/src/network_ops.rs [hunk 5]
```diff
@@ -15,6 +17,15 @@ use secstr::SecUtf8;
 
 /// Interface for performing network operations on Crypto.com Chain
 pub trait NetworkOpsClient {
+    /// fetch current StakedState
+    fn get_staked_state_account(
```
Comment (tomtau) [inline client-network/src/network_ops.rs]: not related to coin serialisation

File: client-network/src/network_ops/default_network_ops_client.rs [hunk 6]
```diff
@@ -8,41 +10,69 @@ use chain_core::tx::data::attribute::TxAttributes;
...
 }
 
-impl<'a, W, S> NetworkOpsClient for DefaultNetworkOpsClient<'a, W, S>
+impl<'a, W, S, C> NetworkOpsClient for DefaultNetworkOpsClient<'a, W, S, C>
 where
     W: WalletClient,
     S: Signer,
+    C: Client,
 {
+    fn get_staked_state_account(
```
Comment (tomtau) [inline client-network/src/network_ops/default_network_ops_client.rs]: not related to coin serialisation

File: client-rpc/src/wallet_rpc.rs [hunk 7]
```diff
@@ -20,7 +20,7 @@ pub trait WalletRpc {
     fn addresses(&self, request: WalletRequest) -> jsonrpc_core::Result<Vec<String>>;
 
     #[rpc(name = "wallet_balance")]
-    fn balance(&self, request: WalletRequest) -> jsonrpc_core::Result<Coin>;
+    fn balance(&self, request: WalletRequest) -> jsonrpc_core::Result<String>;
```
Comment (tomtau) [inline client-rpc/src/wallet_rpc.rs]: any reason why not change the serde implementations as in @calvinaco 's old PR https://github.com/crypto-com/chain/pull/122/files#diff-8fb93dcdc2ace21f62aa8ef871ba2102 ?

**Comment by calvinaco** [comment]
@leejw51crypto  I prefer to change the (De)Serialization of Coin because
- in the future we will introduce new RPC call, if we change `Coin` we will ensure the serialization is applied automatically elsewhere
- If we provide other means of API, the serialization will also work.

File: client-rpc/src/wallet_rpc.rs (hunk 7 above)
Comment (leejw51crypto) [reply client-rpc/src/wallet_rpc.rs]: i'll start from calvain' pr again.

**Comment by tomtau** [comment]
closing this PR, as it'll be started off from the previous one: https://github.com/crypto-com/chain/pull/168#discussion_r299295941

**Pull request closed without merging**